### PR TITLE
Handle NPE if incorrect argument is provided

### LINKED ISSUE #24937: NPE when passing incorrect named arg to the main function via the CLI
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    
}
```

When run as `$ ballerina run zxa.bal -i=1`, with an incorrect named arg;

```log
[2020-07-25 12:34:24,781] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.ballerinalang.jvm.util.ArgumentParser.getEntryFuncArgs(ArgumentParser.java:117)
	at org.ballerinalang.jvm.util.ArgumentParser.extractEntryFuncArgs(ArgumentParser.java:79)
	at ___init.main(.)
```

**Affected Versions:**
slp2

### PR DESCRIPTION
**Pull request by suganyasuven**
Fixes #24937

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/ArgumentParser.java [hunk 1]
```diff
@@ -114,6 +114,9 @@
             if (isNameArg) {
                 isNamedArgFound = true;
                 ParamInfo info = namedArgs.get(getParamName(arg));
+                if (info == null) {
+                    throw BallerinaErrors.createError("incorrect arguments to call the 'main' function");
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/ArgumentParser.java:118]: Shall we include which arg cause the issue. @MaryamZi @praneesha Any thoughts on the error message. BTW fix looks fine for me.

Comment (MaryamZi) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/ArgumentParser.java:118]: Yeah, we usually try to keep these errors similar to the compilation errors on function calls, right? Shall we say something like

`undefined parameter 'param_name'`

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/main/function/ArgumentParserNegativeTest.java [hunk 2]
```diff
@@ -69,6 +69,20 @@ public void testInsufficientArgs() {
         Assert.fail("insufficient arguments not identified");
     }
 
+    @Test
```
Comment (pramodya1994) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/main/function/ArgumentParserNegativeTest.java:72]: Shall we add a `description` for this test case?

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/ArgumentParser.java (hunk 1 above)
Comment (praneesha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/ArgumentParser.java:118]: +1 Better if we can make it "undefined parameter: 'param_name'" only if it doesn't create any inconsistencies with other similar occurrences.

**Pull request merged**